src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerAttributeBindingProvider.cs

```diff
@@ -107,7 +107,7 @@ public CosmosDBTriggerAttributeBindingProvider(INameResolver nameResolver, Cosmo
                 throw new InvalidOperationException(string.Format("Cannot create Collection Information for {0} in database {1} with lease {2} in database {3} : {4}", attribute.CollectionName, attribute.DatabaseName, attribute.LeaseCollectionName, attribute.LeaseDatabaseName, ex.Message), ex);
             }
 
-            return new CosmosDBTriggerBinding(parameter, documentCollectionLocation, leaseCollectionLocation, leaseHostOptions);
+            return new CosmosDBTriggerBinding(parameter, documentCollectionLocation, leaseCollectionLocation, leaseHostOptions, attribute.RetryCount);
```

Comment:
Check at some point, either here or in the Binding or Listener, for values lower than -1. Technically since it's an `int`, the user could input negative values, maybe default to `0` if any negative value is set.

Reply:
The retry code would treat any negative value other than -1 (meaning "infinite") as effectively zero, but I agree it would be nice to notify the customer of a potential typo.  Should I just throw an ArgumentException in this case?

Reply:
Will go the ArgumentException route for now and iterate, pending additional guidance.

Reply:
Correction: InvalidOperationException, following the existing examples in CosmosDBTriggerAttributeBindingProvider